tbmalt/physics/dftb/__init__.py

```diff
@@ -708,48 +710,45 @@ def reset(self):
     from tbmalt import Basis, Geometry
     torch.set_default_dtype(torch.float64)
+    torch.set_printoptions(15)
     from ase.build import molecule
 
     geom = Geometry.from_ase_atoms(molecule('CH4'))
-    basis = Basis(geom.atomic_numbers, shell_dict={1: [0], 6: [0, 1]}, shell_resolved=True)
+    basis = Basis(geom.atomic_numbers, shell_dict={1: [0], 6: [0, 1]})
 
-    path = '../../../tests/unittests/data/io/skfdb.hdf5'
+    path = '../../../examples/example_01/example_dftb_parameters.h5'
     h_feed = ScipySkFeed.from_database(path, [1, 6, 8], 'hamiltonian')
     s_feed = ScipySkFeed.from_database(path, [1, 6, 8], 'overlap')
 
     o_feed = SkfOccupationFeed.from_database(path, [1, 6, 8])
+
     dftb = Dftb1(h_feed, s_feed, o_feed, filling_temp=0.0036749324)
     dftb(geom, basis)
 
+    # DFTB2
     from tbmalt.physics.dftb.feeds import SkFeed, HubbardFeed
-
-    ch4 = torch.tensor([4.30537894059011, 0.92365526485247, 0.92365526485247,
-                        0.92365526485247, 0.92365526485247])
-    h2o = torch.tensor([6.58558984371061, 0.70720507814469, 0.70720507814469])
-
-    geos = Geometry.from_ase_atoms(molecule('H2O'))
-    geob = Geometry.from_ase_atoms([molecule('CH4'), molecule('H2O')])
-    basiss = Basis(geos.atomic_numbers, shell_dict={1: [0], 6: [0, 1], 8: [0, 1]}, shell_resolved=True)
-    basisb = Basis(geob.atomic_numbers, shell_dict={1: [0], 6: [0, 1], 8: [0, 1]}, shell_resolved=True)
     h_feed = SkFeed.from_database(path, [1, 6, 8], 'hamiltonian')
     s_feed = SkFeed.from_database(path, [1, 6, 8], 'overlap')
 
     o_feed = SkfOccupationFeed.from_database(path, [1, 6, 8])
     u_feed = HubbardFeed.from_database(path, [1, 6, 8])
-    dftb2 = Dftb2(h_feed, s_feed, o_feed, u_feed, filling_temp=0.0036749324)
-    dftb2(geos, basiss)
-    assert torch.allclose(dftb2.q_final_atomic, h2o)
 
-    dftb2 = Dftb2(h_feed, s_feed, o_feed, u_feed, filling_temp=0.0036749324)
+    ch4 = torch.tensor([4.30537894059011, 0.92365526485247, 0.92365526485247,
+                        0.92365526485247, 0.92365526485247])
+    h2o = torch.tensor([6.58558984371061, 0.70720507814469, 0.70720507814469])
+
+    geos = Geometry.from_ase_atoms(molecule('CH3O'))
+    geob = Geometry.from_ase_atoms([
+        molecule('H2O'), molecule('CH4'), molecule('CH3O'), molecule('OCHCHO'),
+        molecule('CH3CHO'), molecule('CH3CH2OCH3'), molecule('bicyclobutane')])
+    basiss = Basis(geos.atomic_numbers, shell_dict={1: [0], 6: [0, 1], 8: [0, 1]})
+    basisb = Basis(geob.atomic_numbers, shell_dict={1: [0], 6: [0, 1], 8: [0, 1]})
+
+    mix_params = {'mix_param': 0.2, 'init_mix_param': 0.2, 'generations': 3, 'tolerance': 1e-10}
+    dftb2 = Dftb2(h_feed, s_feed, o_feed, u_feed, filling_temp=0.0036749324, mix_params=mix_params)
     dftb2(geos, basiss)
 
-    dftb2(geob, basisb)
-    assert torch.allclose(dftb2.q_final_atomic[1, :3], h2o)
-    assert torch.allclose(dftb2.q_final_atomic[0, :5], ch4)
-    dftb2(geob, basisb)
```

Comment:
We can probably remove the whole `if __name__ == '__main__':` section from this file